### DISCUSSION TITLE
Access to Menu and button

### DISCUSSION THREAD

**Discussion by akashramtohul4** [quasarframework/quasar#10673] [CLI - PWA mode]
Hello Friends , 

I want to build a scaffolding page with menu and button , I dont know if the team **scaffolding**  is the right word to use. I will try to explain it in diagram .Am looking for advice about how to build it and the right components to use. I dont have any code to show right now 

below is the diagram about how it should look  :

![image](https://user-images.githubusercontent.com/84211129/133328388-3337b3e4-e3fb-4faa-9506-b56efea2eb73.png)

1.**User**  represent the page toggle and just below we have a button toggle **Create** and  **Save** 

I can collapse and open the page toggle and if I select a button on the page toggle should also be selected 

Can someone please advice how I can implement this 

Thanks

**Comment by pikil** [comment]
q-toggle inside q-expansion-item should do the trick I believe :)


https://quasar.dev/vue-components/expansion-item#controlling-expansion-state
https://quasar.dev/vue-components/button

**Reply by akashramtohul4** [reply]
Hello @pikil 

I have been able to implement the expansion with toggle but am facing some issue like "Question 1" :how do i visually show that an expansion have have been selected.

"Question 2" : how do i display the toggle inside my expansion horizontal .

Below is a visual representation on how it look like 

![image](https://user-images.githubusercontent.com/84211129/135907851-7a4350f5-e779-4c2d-95b0-ee748bc33e60.png)

Here Is a section on the code : 
![image](https://user-images.githubusercontent.com/84211129/135911525-72334db4-85df-4ce9-b923-a3fb0ac637b5.png)

I try to replicate in on codepen but its not complicated to make it same 
https://codepen.io/akashramtohul4/pen/vYZqYEJ?editors=1011

Can you please help me

**Reply by pikil** [reply]
Q1: I would suggest doing something similar to this:

```
<div id="q-app">
  <div class="q-pa-md" style="max-width: 550px">
    <q-list bordered class="rounded-borders" v-for="Menu in this.MenuList" :key="Menu.name">
      <q-expansion-item expand-separator icon="perm_identity" :label="Menu" caption="John Doe" v-model="Menu.expanded">
        <template #header>
          <div class="full-width row items-center">
            <div class="col-auto q-pr-md">
              <q-toggle :value="Menu.expanded" />
            </div>
            <div class="col">Users</div>
          </div>
        </template>
        <q-card>
          <q-card-section>
            <q-toggle v-model="value" label="Expanded" ></q-toggle>
            <q-toggle v-model="value" label="Expanded" ></q-toggle>
            <q-toggle v-model="value" label="Expanded" ></q-toggle>
          </q-card-section>
        </q-card>
      </q-expansion-item>
    </q-list>
  </div>
</div>
```

```
new Vue({
  el: '#q-app',
    data () {
    return {
      value: true,
      MenuList:[
        {name: 'test1', expanded: false},
        {name: 'test2', expanded: false},
        {name: 'test3', expanded: false},
        {name: 'test4', expanded: false}
      ]
    }
  }
})
```

Q2: You need to provide a binding to the separate value v-model for each single toggle you have on the page

**Reply by akashramtohul4** [reply]
Yes its an array of object just like MenuList but still am not able to display it side by side

**Reply by pikil** [reply]
Not sure, what you mean by side-by-side.
The toggles inside are side-by-side in your codepen already and all the item-extensions are stacked as per your initial screenshot

**Reply by akashramtohul4** [reply]
what i mean is like this 

![image](https://user-images.githubusercontent.com/84211129/135925038-dd2dca6d-a8ef-4260-a53a-b3836cca1d74.png)

**Reply by pikil** [reply]
![Screen Shot 2021-10-05 at 9 42 30 AM](https://user-images.githubusercontent.com/26779040/135925591-257d99f6-9cc1-4992-afab-a44a06346d53.png)
That's exactly what I see in the codepen you sent :)

Anyhow, you can try to wrap them into the grid row if you want them inline:
https://quasar.dev/layout/grid/row

**Reply by akashramtohul4** [reply]
thanks @pikil you have been a great help :)

**Comment by hawkeye64** [comment]
Is this what you want? https://codepen.io/Hawkeye64/pen/abwgbjW?editors=1011

**Reply by akashramtohul4** [reply]
@hawkeye64  no its sorry i mean horizontal  an how do i indicate that the expansion has been selected. is it possible to add a toggle beside the expansion header ?

**Comment by hawkeye64** [comment]
You can do whatever you want, after all, it's just JavaScript. Here's another try at what "I think" you are asking for. It could be a bit more eloquent: https://codepen.io/Hawkeye64/pen/abwgbjW?editors=1011
